lib/datadog/core/configuration/option.rb

```diff
@@ -28,7 +28,8 @@ module Precedence
             def <=>(other)
               return nil unless other.is_a?(Value)
 
-              numeric <=> other.numeric
+              # Steep does not handle well Structs (https://github.com/ruby/rbs/blob/master/docs/data_and_struct.md)
+              numeric <=> other.numeric # steep:ignore NoMethod
```

Comment:
I've typed structs, not sure what is the issue here? If you define methods or include the module it should work as structs just class definitions